### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by Caren-Koroeny**
<h1>Hello :wave:  Code Reviewer</h1>

> In this pull request I managed to achieve the following as per requirement:

<h2> Add the following validations:</h2>

<h3>For the User model:</h3>

- Name must not be blank.

- PostsCounter must be an integer greater than or equal to zero.

<h3>For the Post model:</h3>
- Title must not be blank.

- Title must not exceed 250 characters.

- CommentsCounter must be an integer greater than or equal to zero. 

- LikesCounter must be an integer greater than or equal to zero.

- Add unit specs for all of your models' methods and validations.

<h1> Knowledge :brain:  gained  :deciduous_tree: </h1>

:cherry_blossom: Use validations for models.

:cherry_blossom: Write unit tests for models.

<h2>Happy Reviewing :pray: </h2>

### CODE REVIEW

File: spec/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  it 'User should have a photo' do
+    expect(subject.photo).to be_present
+  end
+
+  it 'User should have a bio' do
+    expect(subject.bio).to be_present
+  end
+end
```
Comment (thecodechaser) [inline spec/user_spec.rb]: - [ ] Please make sure to add a test for the `returns_last_three_post` method in user model.

File: spec/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  it 'expects post title to be invalid' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'Likes Counter attribute should be greater or equal to zero' do
+    subject.posts_counter = -1
+    expect(subject).to_not be_valid
+  end
```
Comment (thecodechaser) [inline spec/user_spec.rb:17]: - [x] Please make sure to fix the title of the these tests.

File: spec/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  it 'User post counter to increment' do
+    subject.send(:update_posts_counter)
+    expect(subject.author.posts_counter).to be(1)
+  end
+
+  it 'Likes Counter attribute should be greater or equal to zero' do
+    subject.likes_counter = -1
+    expect(subject).to_not be_valid
+  end
```
Comment (thecodechaser) [inline spec/post_spec.rb]: - [ ] Post spec is missing most of the tests for the post model, Please make sure to add tests for all methods and validations described in the post model.

Below are the validations and methods
image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Caren-Koroeny👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Gitflow is followed. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @Caren-Koroeny 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**